app/models/retirement_manager.rb

```diff
@@ -0,0 +1,12 @@
+class RetirementManager
+  def self.check
+    ems_ids = MiqServer.my_server.zone.ext_management_system_ids
+    [LoadBalancer, OrchestrationStack, Vm, Service].each do |model|
```

Comment:
I understand the benefits, but is there actually a nice way to do it?

It is achievable, but the code would be 🙈.